### PR TITLE
Voice changer and mimic voice buffs

### LINKED ISSUE #10264: Voice changing mechanics do not change race.
As the title states, using a voice changer or the changeling's voice mimic does not allow the user to change his or her voice to that of another race, i.e. humans cannot (effectively) pretend to be lizards over comms, and lizards cannot pretend to be humans.

In accordance with coderbus' advice, I request, via issue report, that this be remedied in the interest of promoting equal-opportunity traitorous scum.

### PR DESCRIPTION
**Pull request by Incoming5643**
Mimic voice can now choose a race accent to use.
Voice changers will use the accent associated with the ID.
Agent IDs can now set the species (and thus the accent) associated with it.
IDs given to SUBHUMAN SCUM will have a line generated in examine to show off that fact.

Fixes #10264

Probably others

### CODE REVIEW

**Comment by Rinoka** [comment]
Don't forget to add a way for HoP to stamp the ID as Subhuman Scum.

I approve of this 100% though.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 1]
```diff
@@ -93,6 +94,8 @@
 
 /obj/item/weapon/card/id/examine(mob/user)
 	..()
+	if(registered_race && registered_race.id != "human")
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/cards_ids.dm]: remove the "not human" part, there's no reason to hide that it's a human id, if anything it adds to the specisim RP by being able to brag about the human tag on your ID.

Comment (Incoming5643) [inline code/game/objects/items/weapons/cards_ids.dm]: (I was trying to hide my fuck ups if somehow an id was generated without a species)

I'll see what I can do.

File: code/game/objects/items/weapons/cards_ids.dm [hunk 2]
```diff
@@ -157,6 +160,14 @@ update_label("John Doe", "Clowny")
 			src.registered_name = ""
 			return
 		src.assignment = u
+		var/list/choosable_races = list()
+		for(var/speciestype in typesof(/datum/species) - /datum/species)
+			var/datum/species/S = new speciestype()
```
Comment (RemieRichards) [inline code/game/objects/items/weapons/cards_ids.dm]: don't New these.

1. Isn't there a global list of species (think the init_subtypes proc) to grab from
2. your're only using the id, which can be obtained from the path with ```choosable_races += initial(speciestype.id)``` (this does mean extending the path of speciestype to be absolute though, /datum/species/speciestype)

There's no point or need of newing these, it's a waste.

File: code/game/gamemodes/changeling/powers/mimic_voice.dm [hunk 3]
```diff
@@ -19,6 +20,14 @@
 	var/mimic_voice = stripped_input(user, "Enter a name to mimic.", "Mimic Voice", null, MAX_NAME_LEN)
 	if(!mimic_voice)
 		return
+	var/list/choosable_races = list()
+	for(var/speciestype in typesof(/datum/species) - /datum/species)
```
Comment (RemieRichards) [inline code/game/gamemodes/changeling/powers/mimic_voice.dm]: see below.

File: code/game/objects/items/weapons/cards_ids.dm (hunk 1 above)
Comment (RemieRichards) [inline code/game/objects/items/weapons/cards_ids.dm]: well if an id was generated without a species, so was it's owner right? which... is a much bigger issue that you'd potentially be sweeping under the rug.

Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/cards_ids.dm]: What about the thirty-odd blank ID cards that spawn on the station? What assigned species do they have?

**Comment by Incoming5643** [comment]
As far as I can tell there isn't a global list of species. There's an associated list, species_list, that compares id values against paths, but there's no list that actually contains initialized species datum.

I mean I could make one, but then I'd have to change anything that uses species_list to work with that instead.

**Comment by RemieRichards** [comment]
@Incoming5643 still got to atleast stop ```New()```ing the species datums needlessly, you're only using the ID strings which ```initial()``` can grab without creating the datum,

**Comment by Incoming5643** [comment]
I seem to remember trying to do that and having it not work, but that was two weeks ago and now I've forgotten why.

I'll get back to this eventually at some point.

**Comment by Jordie0608** [comment]
@Incoming5643 Are you still working on this?

**Pull request closed without merging**